### ISSUE TITLE
Europass support

### ISSUE THREAD

**Issue by raisty** [jsonresume/resume-schema#280]
http://interop.europass.cedefop.europa.eu/data-model/

**Comment by jpierson** [comment]
What is meant by support europass within resume-schema directly. To me these seem to be separate competing standards.

**Comment by raisty** [comment]
Support Europass API.
We use it as standard here in Europe.

**Comment by villasv** [comment]
What do you mean with API? Does the Europass standard defines an API? It seems just a document spec to me.

**Comment by stp-ip** [comment]
Yeah it's sorta a competing standard. It might make sense to learn from it or make migrations between the 2 possible so. Also never even seen someone use Europass and I'm a native European...

**Comment by villasv** [comment]
@stp-ip I've seen it frequently used by institutions prospecting international students, usually MBAs and Business Schools. I guess most HR companies still prefer smaller and concise resumes.

**Comment by akarsh** [comment]
@raisty 
I like the idea since I live in Berlin, Germany I had used Europass CV before but now I use JSON Resume and would like to have a CV in the theme of Europass CV.
I saw this project on GitHub https://github.com/jdlopez/europass and asked the project owner on make it clear in the README.md. I upvote for this feature ⬆️

**Comment by stp-ip** [comment]
Would be interesting to know, what made you switch from europass. What did you like about europass and what were the problems?

**Comment by raisty** [comment]
Europass is standard in European Union. Any business even HR should accept it (depends on internal workflow, setup etc.).
I never use for myself something else. Even abroad.

**Comment by akarsh** [comment]
@stp-ip 
In academic jobs in Berlin, the Europass CV makes it a good academic CV.
But when I applied for startup companies and medium companies in Berlin. The HR rejected my CV being too lengthy more than 2 pages. So I had to abandon the Europass CV and create my own Résumés of one page and two pages. I found Resume JSON easy to handle as my data can be changed in the JSON file and I can choose the templates created by other contributors and change them according to my requirements. I published my version of resume theme in npm https://www.npmjs.com/package/jsonresume-theme-onepage-simplified

@raisty 
I agree with you but some companies are rejecting Europass CV I don't know why as they don't mention the reason in the reply emails. Some companies e.g. Mercedes-Benz in Germany have their own portal and have to manually fill all the fields and upload a CV.

**Comment by raisty** [comment]
> @raisty
> I agree with you but some companies are rejecting Europass CV I don't know why as they don't mention the reason in the reply emails. Some companies e.g. Mercedes-Benz in Germany have their own portal and have to manually fill all the fields and upload a CV.

I cannot agree. But understand special needs of HRs. If the company or HR (in European union) don't want accept Europass (as kind of standard), then is something wrong with HR department. Seriously. I work longer time in organisation that cover HR needs for Slovak government and we always looking for standardisations and specific need of HR companies and investors inside Slovakia and connected with Slovakia.
Europass is standard in European union and with support of - for example cover letter - can be wide accepted.
Maybe your issue was caused with missing other documents, but I don't believe the problem is with standardisation of Europass.

I am here to help you.
If you have this issue, feel free contact me and I will do my best to find out what is the problem exactly. I will contact the specific company as well and get the details about issue.
I am not allowed to do some specific tasks regards to your protection.
Let me assist you as best as I can.

**Comment by antialias** [comment]
A request for some type of interoperability (i.e. "support") for Europass is reasonable given that the Europass system is [supported and funded by the EU](https://eur-lex.europa.eu/legal-content/EN/TXT/?uri=celex:32004D2241), appears to be [used by tens of millions of Europeans _per year_ since it launched in 2005](https://europass.cedefop.europa.eu/sites/default/files/statistics_country_sincestart_1.pdf), is backed by [a well defined schema](https://interop.europass.cedefop.europa.eu/sites/default/files/2018-12/europass-xml-schema-doc-v3.3.0.pdf), and even appears to have had [its own magazine that was published for a couple of years](https://europass.cedefop.europa.eu/about/magazine). Also, interoperability with Europass CV would open up a world of possibilities for jsonresume.

This issue is actually two requests, each can be implemented separately:

1. ability to convert _from_ Europass CV _to_ jsonresume.
1. ability to convert _from_ jsonresume _to_ Europass CV.

## For _from_ Europass CV:
Europass CV interop can be implemented as a transformation _from_ a document that validates according to [the Europass JSON schema](https://interop.europass.cedefop.europa.eu/sites/default/files/2018-12/europass-json-schema-v3.3.0.json) _to_ a document with the same content but that validates according to `resume-schema`.

Proposed toolchain:

1. [`europasscv-parser-js`](https://github.com/europass/europasscv-parser-js): to parse a Europass CV (PDF or XML) into a javascript object.
1. either [`json-transforms`](https://github.com/ColinEberhardt/json-transforms), [`jsonpath-object-transform`](https://github.com/dvdln/jsonpath-object-transform), [`jslt`](https://github.com/schibsted/jslt): to transform this object _from_ Europass CV (as a js object) _to_ a jsonresume object. There are a few other json transformation tools reviewed on [this stack overflow thread](https://stackoverflow.com/questions/1618038/xslt-equivalent-for-json).

## For _to_ Europass CV:

Invert the transformation described in the _from_ section. Europass supports JSON natively, so we would be done at that point.

Europass can read resume data embedded inside PDFs so that's also an option for output.

## Where these transformations take place:

These converters can be implemented in their own repositories and used by any number of downstream dependents, ([`resume-cli`](https://github.com/jsonresume/resume-cli) for example.)

Currently resume-cli does not support custom importer / exporters, but its codebase is simple and flexible enough that this wouldn't be a heave lift.

(p.s. I love how the internet lets one reply to a three year old conversation as if it had just happened.)

**Comment by evanplaice** [comment]
@antialias Cool. If this is a tooling-specific issue, should this issue be moved to resume-cli instead?

**Comment by antialias** [comment]
My recommendation in the "Where these transformations take place" section suggests that the impact on resume-cli would only be to implement support for custom importers and exporters (resume-cli issue at https://github.com/jsonresume/resume-cli/issues/356), then implement the _from_ and _to_ pieces as separate packages (perhaps in their own repos). It would be a system similar to how themes are implemented in their own packages. This is all just a proposal so what do you think?
